### ISSUE TITLE
“Expand glob import” generates invalid syntax if the glob is unused

### ISSUE THREAD

**Issue by kpreid** [rust-lang/rust-analyzer#10524]
If a glob import is completely unused, then performing “Expand glob import” will transform valid code such as

```rust
use std::io::*;
```

into

```rust
use std::io::;
```

More correct behavior would be to do one of these:

* Remove the import (closer to canonical warning-free form of source code)
* Generate `use std::io::{};` (more regular, gives a cue as to what is going on rather than “silent” deletion, and will be deleted anyway by rustfmt if format on save is enabled)

---

Context: In the particular case where this came up, I was editing code exclusively targeting `wasm32-unknown-unknown`, and for some reason those files weren't reporting “unused import” warnings at all (this is the first time I tried changing `rust-analyzer.cargo.target` to a cross-compilation target), so the _only_ assist available was to expand the import, not to remove it entirely which would have been a better option.
